### ISSUE TITLE
PositiveDefinite() failed for variance in MultivariateNormal()

### ISSUE THREAD

**Issue by wangyixu14** [TheCamusean/iflow#2]
Hi Julen,

I am encountering another problem when I run python example/train_lasa.py. The __var_ in _conditional_distribution_ function of class _DynamicModel_ in _generic_dynamics.py_ failed the positivedefinite() test for tdist.MultivariateNormal(_mu, _var)(line 113). Do you have any idea about how to fix it?

Thanks,
Yixuan

**Comment by robotgradient** [comment]
Yes. In this implementation I did not add a safe control on the log_var variable, so a priori, it is possible the std to get negative values when training. The easiest implementation is to clip the value to a some positive value.
